panel/widgets/chat.py

```diff
@@ -0,0 +1,1085 @@
+        or anything supported by `pn.pane.Image`. If not set, uses the
+        first character of the name.""")
+
+    reset_on_send = param.Boolean(default=False, doc="""
+        Whether to reset the widget's value after sending a message;
+        has no effect for `TextInput`.""")
+
+    show_rerun = param.Boolean(default=True, doc="""
+        Whether to show the rerun button.""")
+
+    show_undo = param.Boolean(default=True, doc="""
+        Whether to show the undo button.""")
+
+    show_clear = param.Boolean(default=True, doc="""
+        Whether to show the clear button.""")
+
+    _button_data = param.Dict(default={}, doc="""
+        Metadata and data related to the buttons.""")
+
+    _stylesheets: ClassVar[List[str]] = [
+        f"{CDN_DIST}css/chat_interface.css"
+    ]
+
+    def __init__(self, **params):
+        if params.get("widgets") is None:
+            params["widgets"] = [TextInput(placeholder="Send a message")]
+        super().__init__(**params)
+
+        button_icons = {
+            "send": "send",
+            "rerun": "repeat-once",
+            "undo": "arrow-back",
+            "clear": "trash",
+        }
+        for action in list(button_icons):
+            if not getattr(self, f"show_{action}", True):
+                button_icons.pop(action)
+        self._button_data = {
+            name: _ChatButtonData(
+                index=index, name=name, icon=icon, objects=[], buttons=[]
+            ) for index, (name, icon) in enumerate(button_icons.items())
+        }
+        self._input_layout = self._init_widgets()
+        self._composite[:] = [self._chat_log, self._input_layout]
+
+    def _link_disabled_loading(self, obj: Viewable):
+        """
+        Link the disabled and loading attributes of the chat box to the
+        given object.
+        """
+        for attr in ["disabled", "loading"]:
+            setattr(obj, attr, getattr(self, attr))
+            self.link(obj, **{attr: attr})
+
+    def _init_widgets(self) -> Union[Tabs, Row]:
+        """
+        Initialize the input widgets.
+
+        Returns
+        -------
+        The input widgets.
+        """
+        self._widgets = {}
+        for widget in self.widgets:
+            key = widget.name or widget.__class__.__name__
+            if isinstance(widget, type):  # check if instantiated
+                widget = widget()
+            self._widgets[key] = widget
+
+        input_layout = Tabs()
+        for name, widget in self._widgets.items():
+            widget.sizing_mode = "stretch_width"
+            # for longer form messages, like TextArea / Ace, don't
+            # submit when clicking away; only if they manually click
+            # the send button
+            if isinstance(widget, TextInput):
+                widget.param.watch(self._click_send, "value")
+
+            buttons = []
+            for button_data in self._button_data.values():
+                button = Button(
```

Comment:
Please add a description/ tooltip to the Buttons. It will help me understand what the buttons do

![image](https://github.com/holoviz/panel/assets/42288570/76c9eaa7-5f45-4bfb-ac51-6132e7ed8cec)

Reply:
I'd also would like tooltips on button, but not sure if it's doable with panel buttons.

However, I could use the `name` attribute but I feel like the buttons would take up too much space.

Reply:
Added name attributes
<img width="1294" alt="image" src="https://github.com/holoviz/panel/assets/15331990/a3083600-cf60-4c48-ad8e-c9fc204ac0cf">

Reply:
If the icons could be a bit larger it would also help. They are quite small and hard to see.